tools/src/main/scala/org/apache/spark/tools/GenerateMIMAIgnore.scala

```diff
@@ -72,7 +72,10 @@ object GenerateMIMAIgnore {
         val classSymbol = mirror.classSymbol(Class.forName(className, false, classLoader))
         val moduleSymbol = mirror.staticModule(className)
         val directlyPrivateSpark =
-          isPackagePrivate(classSymbol) || isPackagePrivateModule(moduleSymbol)
+          isPackagePrivate(classSymbol) ||
+          isPackagePrivateModule(moduleSymbol) ||
+          classSymbol.isPrivate ||
+          classSymbol.isProtected
```

Comment:
About your previous question, yes, `isPrivate` catches some classes that we did not catch previously (because we were checking for package-private classes, not private classes).